File: mobile/android/focus-android/app/src/androidTest/java/org/mozilla/focus/helpers/MainActivityFirstrunTestRule.java
/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

package org.mozilla.focus.helpers;

import android.content.Context;
import android.preference.PreferenceManager;
import android.support.annotation.CallSuper;
import android.support.test.InstrumentationRegistry;
import android.support.test.rule.ActivityTestRule;

import mozilla.components.browser.session.SessionManager;
import org.mozilla.focus.activity.MainActivity;

import mozilla.components.support.utils.ThreadUtils;
import org.mozilla.focus.ext.ContextKt;

import static org.mozilla.focus.fragment.FirstrunFragment.FIRSTRUN_PREF;

public class MainActivityFirstrunTestRule extends ActivityTestRule<MainActivity> {
    private boolean showFirstRun;

    public MainActivityFirstrunTestRule(boolean showFirstRun) {
        super(MainActivity.class);

        this.showFirstRun = showFirstRun;
    }

    @CallSuper
    @Override
    protected void beforeActivityLaunched() {
        super.beforeActivityLaunched();

        Context appContext = InstrumentationRegistry.getInstrumentation()
                .getTargetContext()
                .getApplicationContext();

        PreferenceManager.getDefaultSharedPreferences(appContext)
                .edit()
                .putBoolean(FIRSTRUN_PREF, !showFirstRun)
                .apply();
    }

    @Override
    protected void afterActivityFinished() {
        super.afterActivityFinished();

        getActivity().finishAndRemoveTask();

        final SessionManager sessionManager = ContextKt.getComponents(
                InstrumentationRegistry.getTargetContext().getApplicationContext()).getSessionManager();

        ThreadUtils.INSTANCE.postToMainThread(new Runnable() {
            @Override
            public void run() {
                sessionManager.removeAll();
            }
        });
    }
}

File: mobile/android/focus-android/app/src/androidTest/java/org/mozilla/focus/helpers/MainActivityFirstrunTestRule.kt
/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

package org.mozilla.focus.helpers

import androidx.annotation.CallSuper
import androidx.preference.PreferenceManager
import androidx.test.platform.app.InstrumentationRegistry
import androidx.test.rule.ActivityTestRule
import mozilla.components.support.utils.ThreadUtils
import org.mozilla.focus.activity.MainActivity
import org.mozilla.focus.ext.components
import org.mozilla.focus.fragment.FirstrunFragment.Companion.FIRSTRUN_PREF

open class MainActivityFirstrunTestRule(launchActivity: Boolean = true, private val showFirstRun: Boolean) :
    ActivityTestRule<MainActivity>(MainActivity::class.java, launchActivity) {

    @CallSuper
    override fun beforeActivityLaunched() {
        super.beforeActivityLaunched()

        val appContext = InstrumentationRegistry.getInstrumentation()
            .targetContext
            .applicationContext

        PreferenceManager.getDefaultSharedPreferences(appContext)
            .edit()
            .putBoolean(FIRSTRUN_PREF, !showFirstRun)
            .apply()
    }

    override fun afterActivityFinished() {
        super.afterActivityFinished()

        ThreadUtils.postToMainThread {
            InstrumentationRegistry
                .getInstrumentation()
                .targetContext
                .applicationContext
                .components
                .tabsUseCases
                .removeAllTabs()
        }
    }
}
